modules/javafx.graphics/src/main/java/javafx/scene/paint/PhongMaterial.java

```diff
@@ -454,51 +803,49 @@ NGPhongMaterial getNGMaterial() {
+            pMaterial.setDiffuseColor(
+                    getDiffuseColor() == null ? null : Toolkit.getPaintAccessor().getPlatformPaint(getDiffuseColor()));
         }
         if (specularColorDirty) {
             pMaterial.setSpecularColor(getSpecularColor() == null ? null
                     : Toolkit.getPaintAccessor().getPlatformPaint(getSpecularColor()));
         }
         if (specularPowerDirty) {
-            pMaterial.setSpecularPower((float)getSpecularPower());
+            pMaterial.setSpecularPower((float) getSpecularPower());
         }
         if (diffuseMapDirty) {
-            pMaterial.setDiffuseMap(getDiffuseMap()
-                    == null ? null : Toolkit.getImageAccessor().getPlatformImage(getDiffuseMap()));
+            pMaterial.setDiffuseMap(
+                    getDiffuseMap() == null ? null : Toolkit.getImageAccessor().getPlatformImage(getDiffuseMap()));
         }
         if (specularMapDirty) {
-            pMaterial.setSpecularMap(getSpecularMap()
-                    == null ? null : Toolkit.getImageAccessor().getPlatformImage(getSpecularMap()));
+            pMaterial.setSpecularMap(
+                    getSpecularMap() == null ? null : Toolkit.getImageAccessor().getPlatformImage(getSpecularMap()));
         }
         if (bumpMapDirty) {
-            pMaterial.setBumpMap(getBumpMap()
-                    == null ? null : Toolkit.getImageAccessor().getPlatformImage(getBumpMap()));
+            pMaterial.setBumpMap(
+                    getBumpMap() == null ? null : Toolkit.getImageAccessor().getPlatformImage(getBumpMap()));
         }
         if (selfIlluminationMapDirty) {
-            pMaterial.setSelfIllumMap(getSelfIlluminationMap()
-                    == null ? null : Toolkit.getImageAccessor().getPlatformImage(getSelfIlluminationMap()));
+            pMaterial.setSelfIllumMap(getSelfIlluminationMap() == null ? null
+                    : Toolkit.getImageAccessor().getPlatformImage(getSelfIlluminationMap()));
         }
 
         setDirty(false);
     }
 
-    @Override public String toString() {
-        return "PhongMaterial[" + "diffuseColor=" + getDiffuseColor() +
-                ", specularColor=" + getSpecularColor() +
-                ", specularPower=" + getSpecularPower() +
-                ", diffuseMap=" + getDiffuseMap() +
-                ", specularMap=" + getSpecularMap() +
-                ", bumpMap=" + getBumpMap() +
-                ", selfIlluminationMap=" + getSelfIlluminationMap() + "]";
+    @Override
+    public String toString() {
+        return "PhongMaterial[" + "diffuseColor=" + getDiffuseColor() + ", specularColor=" + getSpecularColor()
+                + ", specularPower=" + getSpecularPower() + ", diffuseMap=" + getDiffuseMap() + ", specularMap="
+                + getSpecularMap() + ", bumpMap=" + getBumpMap() + ", selfIlluminationMap=" + getSelfIlluminationMap()
+                + "]";
```

Comment:
could we restore the formatting please?
also, the first `+` is unnecessary

Reply:
Weird, I specifically reverted the auto-formatting, not sure how it got in. Will revert.

Reply:
> also, the first `+` is unnecessary

I think it was done for uniformity with the other properties. I would use a `StringJoiner` these days for such a task anyway.

Reply:
A `StringJoiner` seems suitable to me in this case:
```java
new StringJoiner(",", "PhongMaterial[", "]")
    .add("diffuseColor=" + getDiffuseColor())
    .add("specularPower=" + getSpecularPower())
    ...
```
I guess you can make a helper method that takes a property and creates the string for you, like `diffuseColor.getName() + "=" + diffuseColor.getValue()` and then it won't look like a key-value pair in the joiner.

You might want to search for the default implementation of records and see how they create the `name = value` representation.

Reply:
It looks like compiler auto-generates code for records.

The last solution is malloc galore: multiple StringBuilders, much worse than the original concatenation.

All right, sorry for discussion unrelated to this PR.